core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -772,6 +763,17 @@ DETRAY_HOST_DEVICE bool detray::rk_stepper<
     // Advance jacobian transport
     stepping.advance_jacobian(cfg);
 
+    const scalar step_size_scaling = static_cast<scalar>(
+        std::clamp(math::pow(cfg.rk_error_tol / error, 1. / 4.), 0.25, 4.));
+
+    if (navigation() > 0) {
+        stepping._step_size =
+            math::min(stepping._step_size * step_size_scaling, navigation());
```

Comment:
The scale factor has already been applied previously

Reply:
It's for the next step where the `initialized == false`

Reply:
I am confused, will the scale factor then be applied twice if ```initialized == true```

Reply:
It's not going to be applied twice. Please look at where the `advance_track` is called